tests/bsim/README.md

```diff
@@ -0,0 +1,72 @@
+This folder contains tests meant to be run with BabbleSim's physical layer
+simulation, and therefore cannot be run directly from twister.
+
+The compile.sh and run_parallel.sh scripts are used by the CI system to build
+the needed images and execute these tests in batch.
+
+You can also run them manually if desired, but be sure to call them setting
+the variables they expect. For example, from Zephyr's root folder, you can run:
+
+```
+WORK_DIR=${ZEPHYR_BASE}/bsim_out tests/bsim/compile.sh
+RESULTS_FILE=${ZEPHYR_BASE}/myresults.xml SEARCH_PATH=tests/bsim tests/bsim/run_parallel.sh
+```
+
+Or to run only a specific subset, e.g. host advertising tests:
+
+```
+WORK_DIR=${ZEPHYR_BASE}/bsim_out tests/bsim/bluetooth/host/compile.sh
+RESULTS_FILE=${ZEPHYR_BASE}/myresults.xml \
+SEARCH_PATH=tests/bsim/bluetooth/host/adv tests/bsim/run_parallel.sh
+```
+
+Check the run_parallel.sh help for more options and examples on how to use this script to run these
+tests in batch.
+
+After building the tests' required binaries you can run a test directly with its individual
+test scripts.
+
+For example you can build the required binaries for the networking tests with
+
+```
+WORK_DIR=${ZEPHYR_BASE}/bsim_out tests/bsim/net/compile.sh
+```
+and then directly run one of the tests:
+
+```
+tests/bsim/net/sockets/echo_test/tests_scripts/echo_test_802154.sh
+```
+
+
+Conventions about these tests scripts
+-------------------------------------
+
+Each test is defined by a shell script with the extension `.sh`.
+Scripts starting with an underscore (`_`) are ignored.
+
+Each of these test scripts expects the binaries their require already built, and will execute
+the required simulated devices and physical layer simulation with the required command line
+options.
+
+Tests must return 0 to the invoking shell if the test passes, and not 0 if the test
+fails.
+
+It is recommended to have a single test for each test script.
+
+Each test must have a unique simulation id, to enable running different tests in parallel.
+
+
+Conventions about the build scripts
+-----------------------------------
+
+The build scripts (compile.sh) simply build all the required test and sample applications
+for the tests' scripts placed in the subfolders below.
+
+This build scripts use the common compile.source which provide a function (compile) which calls
+cmake and ninja with the provided application, configuration and overlay files.
+
+To speed up compilation for users interested only in a subset of tests, several compile scripts
+exist in several subfolders, where the upper ones call into the lower ones.
+
+Note that cmake and ninja are used directly instead of the `west build` wrapper as west is not
+required, and some Zephyr users do not use or have west, but still use this build and tests scripts.
```

Comment:
Not sure about this: IMO the scripts should switch to use `west` as that's the only thing we define in the getting started guide: https://docs.zephyrproject.org/latest/develop/getting_started/index.html

If you do not have `west`, then you haven't followed that guide, and that means you are either a veteran (and should be aware of `west`) or perhaps should not attempt to run BabbleSim if you are not capable of following guides :)

Reply:
west build is just a convenience wrapper. So using west build does not change the fact that under the hood you are doing the same. And it is a bit irrelevant for users as the result is the same binary anyhow (the usage is hidden inside the common build script). And note that this is the same case for twister.
About west being optional, that was the agreement since the very beginning.

>or perhaps should not attempt to run BabbleSim if you are not capable of following guides :)

Why not? It is not required for anything at this point.

Reply:
No, but this is (yet) another place where BabbleSim works different than any other tool used in Zephyr: I know that `west` is just a wrapper, but why not follow the convention of what we use and document everywhere else? 

Anyone who started using Zephyr in the last 3 years would never have been exposed to any documentation using `cmake` or `ninja` instead of `west`. 

> Why not? It is not required for anything at this point.

Isn't it though? Can you actually stat using Zephyr (without jumping through massive hoops) without doing a `west update` today? Anyone following the https://docs.zephyrproject.org/latest/develop/getting_started/index.html and getting the dependencies via `pip install -r ~/zephyrproject/zephyr/scripts/requirements.txt` will have `west`.

Reply:
This works just like in twister. And this is only an internal matter for the compilation script.
But why does this matter in this context? Any user can build with west their test/app for a bsim board if they want to.

Reply:
It's fine to leave as is, but I don't think it adds much value. If this is meant for users, then I'd argue it's better to switch to `west` for these scripts in the near future and reference `west` here instead. 

If this is not meant for users, then `some Zephyr users` is just confusing :)